File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
from collections import *
def solution(want, number, discount):
    answer = 0
    l = len(discount)
    zipped = []
    for w,n in zip(want,number): 
        zipped += ([w] * n)
    
    check = Counter(zipped)


    for begin in range(l-9):
        items = discount[begin:begin+10]
        citems = Counter(items)
        
        if(citems == check):
            answer += 1
            
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.swift
import Foundation

func solution(_ want:[String], _ number:[Int], _ discount:[String]) -> Int {
    
    var target : [String:Int] = [:]
    
    for (n,w) in zip(number,want) {
        target[w] = n 
    }
    
    var ans: Int = 0
    
    outer: for i in 0...discount.count-10 {
        
        var curr: [String:Int] = [:]
        var flag: Bool = true 
        
        for j in 0...9 {
            curr[discount[i+j],default:0] += 1 
        }
        
        for k in target.keys {
           if curr[k] == nil { // 현재 세일 품목에 없거나
               flag = false 
               break
           } 
            
           if curr[k]! < target[k]! {  // 현재 세일 품목 갯수가 부족
                flag = false 
               break
           }
        }
        
        if flag {
            ans += 1
        }
        
    }
    
    return ans
}